### DISCUSSION TITLE
What features do you expect from EasyCaching 2.0

### DISCUSSION THREAD

**Discussion by catcherwong** [dotnetcore/EasyCaching#470] [Ideas]
What features do you expect from EasyCaching 2.0

**Comment by catcherwong** [comment]
1. Support Set a cache without expiration
2. Support sliding expiration
3. Event-based DiagnosticSource (For some apm system such as open-telemetry, skyapm, etc.)
4. Add more useful providers, serializers, buses and interceptors to [easycaching-contrib](https://github.com/easycaching-contrib), such as [FreeRedis](https://github.com/2881099/FreeRedis/), [NewLife.Redis](https://github.com/NewLifeX/NewLife.Redis), etc.

**Reply by Karbust** [reply]
Thumbs up for support to set a cache without expiration, was looking on how to do it and found this...

**Comment by catcherwong** [comment]
5. Add TestBase project for other projects

**Comment by rLindorfer** [comment]
6. Set maximum disk size which can be used for caching

**Comment by kaykaul** [comment]
@catcherwong 
Option to provide serializer when setting or getting a value. I'd like to be able to do this, `cacheProvider.Get<TObject>(key, customSerializer)` or `cacheProvider.Get<TObject>(key, serializerName: "customSerializer")`

Or using fluent API like:
`cacheProvider.WithSerializer(customSerializer).Get<TObject>(key)`

**Reply by kaykaul** [reply]
Or with c#'s `using` keyword:
```csharp
using(var cache = cacheProvider.UsingSerializer(customSerializer))
{
    var item = cache.Get<TObject>(key)
}
```

**Comment by artyom-p** [comment]
`.AddEasyCaching()` overload with `IServiceProvider` as a parameter.

**Comment by KANekT** [comment]
I need support net 8.0
